w08h03/test/pgdp/datastructures/UnitTest.java

```diff
@@ -39,7 +39,8 @@ <Type extends Comparable<Type>> void testArray(Type[] input) {
 	@Test
 	@DisplayName("should iterate over empty graph")
 	public void emptyTest() {
-		assertThrows(NoSuchElementException.class, () -> (new QuarternarySearchTree<Integer>()).iterator().next());
+		Exception e = assertThrows(NoSuchElementException.class, () -> (new QuarternarySearchTree<Integer>()).iterator().next());
+		assertEquals(e.getMessage(), "Trying to call next on an empty QuarternarySearchTreeIterator");
```

Comment:
Thanks for improving this ❤️ But for the next time it would be better to split this PR into two PRs, one for adding a new test, one for fixing an existing test 👍

Reply:
Are you sure that the order of `expected` and `actual` is correct in `assertEquals`?

Reply:
Short question: Where is the message "Trying to call next on an empty QuarternarySearchTreeIterator" coming from? Because the test is testing that you throw like this `throw new NoSuchElementException("Trying to call next on an empty QuarternarySearchTreeIterator")`

Reply:
There is a line in the template that you are supposed to use to throw a NoSuchElementException. I missed that too at first and didn't add an Error message at all so I decided to make a test for it.

Reply:
Oh, I see. Extremely good catch! Thanks 🙏